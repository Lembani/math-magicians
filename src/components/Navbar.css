.navbar {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 20px;
  box-shadow: rgba(50, 50, 93, 0.25) 0 30px 60px -12px, rgba(0, 0, 0, 0.3) 0 18px 36px -18px;
}

.brand {
  font-size: 2.5rem;
}

.nav-list-container {
  width: 20%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  list-style: none;
}

.nav-list-item {
  padding: 0 10px;

  font-size: 1.2rem;
}

.nav-link {
  color: rgb(243, 168, 31);
  text-decoration: none;
}

.nav-link:hover {
  color: #000;
}
