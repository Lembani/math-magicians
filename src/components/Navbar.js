import React from 'react';
import { Link } from 'react-router-dom';

import './Navbar.css';

class Navbar extends React.PureComponent {
  render() {
    return (
      <>
        <nav className="navbar">
          <Link className="brand nav-link" to="/">Math Magicians</Link>
          <ul className="nav-list-container">
            <li className="nav-list-item">
              <Link to="/" className="nav-link">Home</Link>
            </li>
            |
            <li className="nav-list-item">
              <Link to="/calculator" className="nav-link">Calculator</Link>
            </li>
            |
            <li className="nav-list-item">
              <Link to="/quote" className="nav-link">Quote</Link>
            </li>
          </ul>
        </nav>
      </>
    );
  }
}

export default Navbar;
