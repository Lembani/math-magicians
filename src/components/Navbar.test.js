import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from './Navbar';

describe('Should be in document and render correctly on the screen', () => {
  test('Nav is in document', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    const navElement = document.querySelector('.navbar');
    expect(navElement).toBeInTheDocument();
  });

  it('Renders to the screen', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
