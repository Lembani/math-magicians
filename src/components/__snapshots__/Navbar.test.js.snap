// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be in document and render correctly on the screen Renders to the screen 1`] = `
<nav
  className="navbar"
>
  <a
    className="brand nav-link"
    href="/"
    onClick={[Function]}
  >
    Math Magicians
  </a>
  <ul
    className="nav-list-container"
  >
    <li
      className="nav-list-item"
    >
      <a
        className="nav-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    |
    <li
      className="nav-list-item"
    >
      <a
        className="nav-link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    |
    <li
      className="nav-list-item"
    >
      <a
        className="nav-link"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
